webapp/channels/src/components/flag_message_modal/flag_post_modal.tsx

```diff
@@ -138,16 +142,24 @@ export default function FlagPostModal({postId, onExited}: Props) {
         return hasError;
     }, [comment, contentFlaggingSettings?.reporter_comment_required, formatMessage, reason]);
 
-    const handleConfirm = useCallback(() => {
+    const handleConfirm = useCallback(async () => {
         const hasError = validateForm();
         if (hasError) {
             return;
         }
 
-        // TODO: Implement the flagging action here in a follow up PR
-
-        onExited();
-    }, [validateForm, onExited]);
+        try {
+            setSubmitting(true);
+            await Client4.flagPost(post.id, reason, comment);
+            setSubmitting(false);
+            onExited();
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error(error);
+            setRequestError((error as ServerError).message);
+            setSubmitting(false);
+        }
```

Comment:
0/5 nitpick

Move the `setSubmitting` false outside of the try/catch block or in a finally statement?

Reply:
Done.